:recycle: Restructuring the Website (#251)
ros-planning/moveit2_tutorials: doc/how_to_contribute/how_to_contribute_to_site.rst

> How-To Contribute to This Site

Reviewer: ```suggestion
How to Contribute to This Site
```

==============================

This explains how to make and submit changes to this site.

> This how-to guide assumes you are on Ubuntu 20.04.

Reviewer: Would a section for "requirements" or "what you need" be good for a standard structure in the how-tos?

Author: I like that.  You could also add that to the tutorials for how to write a how-to?

Setting up your development environment
---------------------------------------

Follow the tutorial :doc:`/doc/tutorials/getting_started/getting_started` to set up your workspace for testing local changes.

All further instructions assume you are the directory containing a local checkout of ``moveit2_tutorials``.

> Build and view the website locally
> ----------------------------------

Reviewer: These seem a bit short to be subsections, maybe just bolded headers?
```suggestion
**Build and view the website locally**
```

To build the website locally run this command:

.. code-block:: bash

  make html

> Then you can use this to open the site in your webbrowser:
> 
> .. code-block:: bash
> 
>   open ./build/html/index.html

Reviewer: ```suggestion
Then you can open the local build of the site in your web browser from: ``/build/html/index.html``. For example, to open the local site in Firefox the command would be:

.. code-block:: bash

  firefox ./build/html/index.html
```

Author: I'd put a . or remove the first slash from the instructions you added.  Does that make sense?  The build directory isn't at the root of the filesystem.

Author: I've included this change.

> Test for broken links
> ---------------------

Reviewer: ```suggestion
**Test for broken links**
```

> To test for broken links run this command:

Reviewer: Is there a time people should be doing this, or is this just a helpful command?

Author: Before they submit their PR they should run both this and the pre-commit change.  This is what is run in CI to test for broken links.

.. code-block:: bash

  ./htmlprofer.sh

> Run the formatters and spellchecker
> -----------------------------------

Reviewer: ```suggestion
**Run the formatters and spellchecker**
```

We use `pre-commit <https://pre-commit.com/>`_ to run the formatting and spelling checkers.
You can install it with pip like this:

.. code-block:: bash

  python3 -m pip install --user pre-commit

To run pre-commit locally to fix formatting and spelling:

.. code-block:: bash

  pre-commit run --all

> Run industrial_ci locally to run CI
> -----------------------------------

Reviewer: ```suggestion
**Run industrial_ci locally to run CI**
```

1. Clone a copy of `industrial_ci <https://github.com/ros-industrial/industrial_ci>`_ into your workspace.
2. Build and source your workspace.
3. Run this command from the workspace directory to test code changes just as it is done in CI:

.. code-block:: bash

  ros2 run industrial_ci rerun_ci src/moveit2_tutorials \
    DOCKER_IMAGE='moveit/moveit2:rolling-source' \
    UPSTREAM_WORKSPACE='moveit2_tutorials.repos' \
    TARGET_CMAKE_ARGS='-DCMAKE_BUILD_TYPE=Release' \
    CCACHE_DIR="$HOME/.ccache" \
    CLANG_TIDY='true'

> Next Step
> ---------
> 
> :doc:`how_to_write_tutorials`

Reviewer: ```suggestion
**Next Step:** :doc:`how_to_write_tutorials`
```
